server.go

```diff
@@ -303,6 +303,8 @@ type Server struct {
 	DecorateReader DecorateReader
 	// DecorateWriter is optional, allows customization of the process that writes raw DNS messages.
 	DecorateWriter DecorateWriter
+	// Maximum number of TCP queries before we close the socket. Default is maxTCPQueries (unlimited if -1).
+	MaxTCPQueries int
```

Comment:
I'm a little hesitant about using a magic number here, but it's not blocking to me. (I still like something like `const UnlimitedMaxQueries = -1` simply for the clarity it gives).